app/services/rbac.rb

```diff
@@ -35,6 +37,23 @@ def load_user_permissions(identity)
       end
     end
 
+    # Scoped for inventory groups
+    def supported_operation?(attribute_filter)
+      attribute_filter[:operation] == 'in' && attribute_filter[:key] == 'groups.id' ||
+        attribute_filter[:operation] == 'equal' && attribute_filter[:key] == 'group.id'
+    end
+
+    # Scoped for inventory groups
+    def operation_value(attribute_filter)
+      if attribute_filter[:operation] == 'in' && attribute_filter[:key] == 'groups.id'
+        # Received '[nil]' symbolizes access to ungrouped entries.
+        # In the output, the ungrouped entries are symbolized with '{}'.
```

Comment:
I thought they are symbolized with empty array...

Reply:
`flatten(1)` should do the trick

Reply:
Or maybe use `flat_map` instead of the `map`

Reply:
```
[5] pry(#<User>)> a
=> [["78e3dc30-cec3-4b49-be2d-37482c74a9ac", "79e3dc30-cec3-4b49-be2d-37482c74a9ad"], "80e3dc30-cec3-4b49-be2d-37482c74a9ad", [], nil, nil]
[6] pry(#<User>)> a.flatten(1)
=> ["78e3dc30-cec3-4b49-be2d-37482c74a9ac", "79e3dc30-cec3-4b49-be2d-37482c74a9ad", "80e3dc30-cec3-4b49-be2d-37482c74a9ad", nil, nil]
```

Using `flatten(1)` for this would still remove the `[]` representing ungrouped hosts.

Reply:
Will try look for alternative.